docs/stamping.md

```diff
@@ -7,38 +7,63 @@ toc: true
 # Stamping
 
 Bazel is generally only a build tool, and is unaware of your version control system.
-However, when publishing releases, you typically want to embed version information in the resulting distribution.
-Bazel supports this natively, using the following approach:
+However, when publishing releases, you may want to embed version information in the resulting distribution.
+Bazel supports this with the concept of a "Workspace status" which is evaluated before each build.
+See [the Bazel workspace status docs](https://docs.bazel.build/versions/master/user-manual.html#workspace_status)
 
 To stamp a build, you must pass the `--stamp` argument to Bazel.
 
-> Previous releases of rules_nodejs stamped builds always.
-> However this caused stamp-aware actions to never be remotely cached, since the volatile
-> status file is passed as an input and its checksum always changes.
+Stamping is typically performed on a later action in the graph, like on a packaging rule (`pkg_*`). This means that
+a changed status variable only causes re-packaging, not re-compilation and thus does not cause cascading re-builds.
 
-Also pass the `workspace_status_command` argument to `bazel build`.
-We prefer to do these with an entry in `.bazelrc`:
+Bazel provides a couple of statuses by default, such as `BUILD_EMBED_LABEL` which is the value of the `--embed_label`
+argument, as well as `BUILD_HOST` and `BUILD_USER`. You can supply more with the workspace status script, see below.
 
-```sh
-# This tells Bazel how to interact with the version control system
-# Enable this with --config=release
-build:release --stamp --workspace_status_command=./tools/bazel_stamp_vars.sh
+Some rules accept an attribute that uses the status variables.
+For example, in a `pkg_npm` you can use the `substitutions` attribute like:
+
+```python
+pkg_npm(
+    name = "npm_package",
+    substitutions = {"0.0.0-PLACEHOLDER": "{STABLE_GIT_COMMIT}"},
```

Comment:
do we need to make it clearer that if `--stamp` isn't set, the placeholder will be left alone, rather than set to empty string? Or what happens if the variable isn't defined?

Reply:
yes, that's a subtle behavior change in this PR actually, we used to replace it with `0.0.0` if there was no stamp info. I'll update the doc and the commit description